duckscript_sdk/src/sdk/std/zip/pack/mod.rs

```diff
@@ -0,0 +1,232 @@
+                    "--append" => append = true,
+
+                    _ => {
+                        zipfile = Some(argument.as_str());
+                        looking_for = LookingFor::Files;
+                    }
+                },
+
+                LookingFor::Compression => {
+                    match argument.as_str() {
+                        "deflate" => compression = CompressionMethod::Deflated,
+                        "bzip2" => compression = CompressionMethod::Bzip2,
+                        "none" => compression = CompressionMethod::Stored,
+
+                        _ => return CommandResult::Error("Unknown compression method.".to_string()),
+                    }
+
+                    looking_for = LookingFor::Options;
+                },
+
+                LookingFor::Base => {
+                    let base_str = argument.as_str();
+                    base = Some(base_str.strip_prefix("./").unwrap_or(base_str));
+                    looking_for = LookingFor::Options;
+                },
+
+                LookingFor::Files => file_args.push(argument.as_str()),
+            }
+        }
+
+        if file_args.is_empty() {
+            return CommandResult::Error("Input files not provided.".to_string());
+        }
+
+        let zipfile = match zipfile {
+            Some(filename) => filename,
+            None => return CommandResult::Error("ZIP file name not provided.".to_string()),
+        };
+
+        let zipfile = Path::new(zipfile);
+        let zipfile_dir = match zipfile.parent() {
+            Some(path) => path,
+            None => return CommandResult::Error("Couldn't parse ZIP file directory.".to_string()),
+        };
+        match std::fs::create_dir_all(zipfile_dir) {
+            Ok(_) => (),
+            Err(err) =>
+                return CommandResult::Error(format!("Couldn't create ZIP file directory: {}", err)),
+        };
+        let zip_file_existed = zipfile.exists();
+        let zip_file = match OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(!zip_file_existed)
+            .truncate(!append)
+            .open(zipfile)
+        {
+            Ok(file) => file,
+            Err(err) =>
+                return CommandResult::Error(format!("Couldn't create/open ZIP file: {}", err)),
+        };
+        let mut zip = if append && zip_file_existed {
+            match ZipWriter::new_append(zip_file) {
+                Ok(writer) => writer,
+                Err(err) =>
+                    return CommandResult::Error(format!("Couldn't open ZIP file: {}", err)),
+            }
+        } else {
+            ZipWriter::new(zip_file)
+        };
+
+        let zip_options = FileOptions::default()
+            .compression_method(compression)
+            .unix_permissions(0o755);
+
+        let mut files = Vec::new();
+        for arg in file_args {
+            match state::get_handle(state, arg.to_string()) {
```

Comment:
worth mentioning in the help text that files can be array handles

Reply:
Updated the help